core/templates/dev/head/domain/learner_dashboard/LearnerDashboardIconsDirective.js

```diff
@@ -0,0 +1,98 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for showing learner dashboard icons.
+ */
+
+oppia.directive('learnerDashboardIcons', [
+  'UrlInterpolationService', function(UrlInterpolationService) {
+    return {
+      restrict: 'E',
+      scope: {
+        getActivityType: '&activityType',
+        getActivityId: '&activityId',
+        getActivityTitle: '&activityTitle',
+        activityActive: '=activityActive'
+      },
+      templateUrl:  UrlInterpolationService.getDirectiveTemplateUrl(
+        '/domain/learner_dashboard/' +
+        'learner_dashboard_icons_directive.html'),
+      controller: [
+        '$scope', 'LearnerDashboardIdsBackendApiService',
+        'LearnerDashboardActivityIdsObjectFactory',
+        'LearnerPlaylistService',
+        function($scope, LearnerDashboardIdsBackendApiService,
+                 LearnerDashboardActivityIdsObjectFactory,
+                 LearnerPlaylistService) {
+          $scope.activityIsActive = true;
+
+          $scope.$watch('activityActive', function(value) {
```

Comment:
Why do you need to $watch -- doesn't it update automatically? Couldn't you just bind $scope.activityIsActive directly to the attr passed in from the parent?

Another approach might be: the parent passes in an attr along the lines of "parent-tile-is-being-hovered-over", and you can use that as an "&" attr. Then you can maintain a local $scope.iconsAreBeingHoveredOver and also compute $scope.activityIsActive based on whether at least one of the previous two things is true. I think it depends on what exactly you want $scope.activityIsActive to represent (and I suggest choosing a better name -- see above -- and possibly also adding a docstring/comment), but this could use some more clarity.

Reply:
Everywhere I've read about this it is advised to use the watch functionality to get this behavior. For example - [https://stackoverflow.com/a/20068832/6331311](https://stackoverflow.com/a/20068832/6331311). Also, I've changed the variable names so hopefully this makes more sense.

Reply:
Hm, OK. I would have thought you could pass them in directly, but I'm not _too_ worried. The concern with $watch is performance -- so maybe try adding a console.log() inside it and see how often it gets called. If it doesn't get called too much then what you have is fine.